x-pack/platform/plugins/shared/lens/public/app_plugin/lens_document_equality.ts

```diff
@@ -45,8 +45,10 @@ export const isLensEqual = (
           return isEqualFromVis(
             doc1.state.visualization,
             doc1.references,
+            doc1.state.datasourceStates?.formBased,
             doc2.state.visualization,
             doc2.references,
+            doc2.state.datasourceStates?.formBased,
```

Comment:
```suggestion
            doc2.state.datasourceStates,
```

Reply:
Fixed in 878a48556a848b8d34945cb7d90713e1a254928b